src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/UMI.java

```diff
@@ -0,0 +1,66 @@
+package org.broadinstitute.hellbender.tools.walkers.mutect;
+
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.broadinstitute.hellbender.engine.filters.UMIReadFilter;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
```

Comment:
This class has lots of unused/untested stuff that probably should be used / tested.  The umi logic should be implemented here and called from the filter.